api/src/game.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { Game, Prisma } from '@prisma/client';
+import { GameState } from './types';
+import { createNewGameState } from './game';
+import { graphToBoard, boardToGraph } from './graph';
+
+@Injectable()
+export class GameService {
+  constructor(private prisma: PrismaService) {}
+
+  async getGame(
+    userWhereUniqueInput: Prisma.GameWhereUniqueInput,
+  ): Promise<Game | null> {
+    return this.prisma.game.findUnique({
+      where: userWhereUniqueInput,
+    });
+  }
+
+  async getGames(params: {
+    skip?: number;
+    take?: number;
+    cursor?: Prisma.GameWhereUniqueInput;
+    where?: Prisma.GameWhereInput;
+  }): Promise<Game[]> {
+    const { skip, take, cursor, where } = params;
+    return this.prisma.game.findMany({
+      skip,
+      take,
+      cursor,
+      where,
+    });
+  }
+
+  async createGame(): Promise<Game> {
+    const newGameState: GameState = createNewGameState();
+
+    const data = {
+      board: JSON.stringify(graphToBoard(newGameState.graph)),
+      currentPlayer: 1,
+      players: JSON.stringify(newGameState.players),
+      hasWinner: false,
+      started: true,
+    };
+    return this.prisma.game.create({ data });
+  }
+
+  async updateGame(params: {
+    where: Prisma.GameWhereUniqueInput;
+    data: Prisma.GameUpdateInput;
+  }): Promise<Game> {
+    const { where, data } = params;
+    return this.prisma.game.update({
+      data,
+      where,
+    });
+  }
+
+  async deleteGame(where: Prisma.GameWhereUniqueInput): Promise<Game> {
+    return this.prisma.game.delete({
+      where,
+    });
+  }
+
+  bddEntryToGameState(bddEntry): GameState {
```

Comment:
bdd means Base de Données and it Frenglish

Also you can call it serializer/deserializer

Reply:
cleaver ! Done